### ISSUE TITLE
maybe_from_gregorian drops nanos parameter

### ISSUE THREAD

**Issue by patricklucas** [ChristopherRabotin/hifitime#85]
While [`Epoch::maybe_from_gregorian`](https://github.com/ChristopherRabotin/hifitime/blob/d2f308840a942861d654c6d3325ccd0bb4452977/src/epoch.rs#L235) validates that the `nanos` parameter is valid (via [`is_gregorian_valid`](https://github.com/ChristopherRabotin/hifitime/blob/d2f308840a942861d654c6d3325ccd0bb4452977/src/epoch.rs#L957)), it does not actually include it in the quantity of seconds used to create the resulting `Epoch`.

I suppose the simple fix would be:

```diff
diff --git a/src/epoch.rs b/src/epoch.rs
index 1623ce8..ae9870c 100644
--- a/src/epoch.rs
+++ b/src/epoch.rs
@@ -265,7 +265,8 @@ impl Epoch {
         seconds_wrt_1900 += TimeUnit::Day * (day - 1)
             + TimeUnit::Hour * hour
             + TimeUnit::Minute * minute
-            + TimeUnit::Second * second;
+            + TimeUnit::Second * second
+            + TimeUnit::Nanosecond * nanos;
         if second == 60 {
             // Herein lies the whole ambiguity of leap seconds. Two different UTC dates exist at the
             // same number of second afters J1900.0.
```

Apologies if there's a reason for limiting this to second resolution that I didn't pick up on.

**Comment by ChristopherRabotin** [comment]
Good catch, I had missed that! Thanks

**Comment by ChristopherRabotin** [comment]
@patricklucas When do you need this change on crates.io? I was planning on doing another release in the next two weeks with some changes to increase the speed of the library, so I could add your fix in that release. If you need it sooner, I can make an intermediate release as well.

**Comment by patricklucas** [comment]
No rush, two weeks is fine. I can use a `git` dependency in the meantime. Thanks!

**Comment by ChristopherRabotin** [comment]
Published ;-)

**Comment by patricklucas** [comment]
Thanks.

Actually it's just a personal project to play with the [HYG database](https://github.com/astronexus/HYG-Database) to plot sky maps for a place and time. I don't remember exactly where, but I ran across hifitime when looking for Rust libraries for doing the Julian time conversions to make use of the J2000 positions in the database.

I found this issue when I tried running it in a loop using the current time—and noticed the alt/az only update once per second!
